### ISSUE TITLE
Bump to etcd v2.2.4

### ISSUE THREAD

**Issue by crawford** [coreos/bugs#1068]

**Comment by gyuho** [comment]
Ok sounds good.

**Comment by crawford** [comment]
We need it this week. This is holding up the entire release pipeline for all channels.

**Comment by gyuho** [comment]
@xiang90 Tomorrow?

**Comment by xiang90** [comment]
@gyuho We need to do some testing around the TLS fix.

@crawford For patch releases, the timeline is about week not day. We are not going to do a release right after we found a regression issues. First fixing the regression takes time, second testing it takes time. Do not expect us to do a release within 3 days after we found it.

**Comment by crawford** [comment]
@xiang90 it looks like the patch has been merged (so I assume all testing has passed). What else needs to be done? I expect a three-day turnaround in this case because we are _still_ waiting on a fix from the last regression. At the end of the day, all I care about is the etcdctl-v0.4 fix for the Stable channel.

**Comment by xiang90** [comment]
@crawford We already release etcd 2.2.3 with etcdctl fix. You can go and release that if that is all you care about.

**Comment by crawford** [comment]
If we bump to 2.2.3, it will introduce this TLS regression, no?

**Comment by xiang90** [comment]
@crawford The TLS regression introduced back to last Aug. We just found it. It is not a regression over 2.2.2 or 2.2.3.

**Comment by crawford** [comment]
In which commit was the regression introduced? I don't see it listed on https://github.com/coreos/etcd/pull/4153.

**Comment by xiang90** [comment]
@crawford https://github.com/xiang90/etcd/commit/97923ca3fc6c3b94a3961f54a21ae76c39e40121

**Comment by crawford** [comment]
OK. We'll go with v2.2.3.

**Comment by xiang90** [comment]
@crawford 

>  it looks like the patch has been merged. What else needs to be done?

It merged into master branch not the release branch.

1. Cherry pick necessary patches into release branch including other patches.

2. Check if it builds on release branch

3. Run CI on release branch

4. Manually test it by etcd team (upgrading from pervious version, basic commands work)

5. Finish other release process like tag, writing docs, and updating tars.
